conanfile.py

```diff
@@ -136,6 +137,22 @@ def _gather_licenses(self, bootstrap):
             f" --with-license-file --no-license-path --output-file=package_licenses.txt"
         )
 
+    def _gather_packages(self):
+        """Gather all the required packages into a file for future reference
+
+        We don't have an easy way to get this so parse the package names out of the requirements
+        file.
+        """
+        requirements = self.make_requirements_file()
+        packages = pathlib.Path("packages.txt").resolve()
+        matcher = re.compile(r"^([\w.-]+)==[\w.-]+$", flags=re.MULTILINE)
```

Comment:
ChatGPT approves this regex:

![Screenshot from 2023-04-17 14-16-44](https://user-images.githubusercontent.com/8601395/232481506-edea90c3-f735-494c-929b-c038abdcdf38.png)

Reply:
I think we shouldn't have commented out requirements anyway if you feel really defensive about it it can be added but feels overkill for such a simple thing.

Reply:
All the requirements that end up in this file will always have exact `==` requirements so this is all good. If the package requirements file that ended up at this stage somehow does not have that, it was a mistake to begin: loose requirements would mean that the embedded Python package would produce different results depending on when it was built, which goes against what we're trying to do with packaging in general.